### PR TITLE
feat(AWS MSK): Support `filterPatterns`

### PR DESCRIPTION
**Pull request by shreyance-jain**
Addresses: #[11635](https://github.com/serverless/serverless/issues/11635)

### CODE REVIEW

File: lib/plugins/aws/package/compile/events/msk/index.js [hunk 1]
```diff
@@ -140,6 +142,14 @@ class AwsCompileMSKEvents {
             mskResource.Properties.SourceAccessConfigurations = secureAccessConfigurations;
           }
 
+          if (filterPatterns != null) {
```
Comment (medikoo) [inline lib/plugins/aws/package/compile/events/msk/index.js:145]: Let's simplify to `if (filterPatterns)`

**Review by medikoo — COMMENTED** [review]
Thank you @shreyance-jain !

File: lib/plugins/aws/package/compile/events/msk/index.js (hunk 1 above)
Comment (shreyance-jain) [reply lib/plugins/aws/package/compile/events/msk/index.js]: Changed, sticking to new format.

**Review by medikoo — CHANGES_REQUESTED** [review]
@shreyance-jain unfortunately unit tests fail, can you check why is the case?

**Comment by shreyance-jain** [comment]
@medikoo Fixed. Should pass now.

**Comment by shreyance-jain** [comment]
Also, @medikoo when I run `npm test` locally, it is not able to pick aws config
Any idea what could be possible issue here

npm-test-error

**Comment by medikoo** [comment]
>  Also, @medikoo when I run npm test locally, it is not able to pick aws config

Can you show full stack trace and isolate to single test ?

**Comment by shreyance-jain** [comment]
Working fine when isolated to single test
isolated-test-result

**Comment by medikoo** [comment]
@shreyance-jain yes it's definitely not a problem introduced with this PR. Still as other users also had similar issue I wonder what triggers that.

Therefore are you able to post a full stack trace and isolate it to single test from our unit tests?

**Review by medikoo — APPROVED** [review]
Thank you @shreyance-jain !

**Comment by shreyance-jain** [comment]
Yeah, here is the full stack trace 

npm-test-error-stack-trace

**Comment by shreyance-jain** [comment]
> Thank you @shreyance-jain !

Thank you very much for quick issue & pr review!

**Comment by medikoo** [comment]
It appears that this environment variable influences AWS SDK v2 import: https://github.com/aws/aws-sdk-js/issues/4043
and currently, our tests are not run isolated against what's set in environment variables (while the home directory is mocked to a temporary directory, which doesn't have `.aws` folder, hence the failure)

**Comment by shreyance-jain** [comment]
> @shreyance-jain I believe you get this error as you have `AWS_SDK_LOAD_CONFIG` env variable set in environment (?)

I unset it & worked 💯 




> It appears that this environment variable influences AWS SDK v2 import: [aws/aws-sdk-js#4043](https://github.com/aws/aws-sdk-js/issues/4043) and currently, our tests are not run isolated against what's set in environment variables (while the home directory is mocked to a temporary directory, which doesn't have `.aws` folder, hence the failure)

Got it! Thank you @medikoo!

**Pull request merged**